tmt/steps/prepare/__init__.py

```diff
@@ -51,6 +51,7 @@ class PrepareStepData(tmt.steps.WhereableStepData, tmt.steps.StepData):
 class _RawPrepareStepData(tmt.steps._RawStepData, total=False):
     where: Optional[list[str]]
     package: list[str]
+    epel: str
```

Comment:
`_RawPrepareStepData` is a shared base for all `prepare` plugins, as such it should not contain plugin-specific keys, which `epel` definitely is. Even `package` and `missing` should be removed, this seems like a mistake which then led you to add `epel` here. Sorry for the mess.

Type linters should not complain when you set the `epel` key in your `data: _RawPrepareStepData` on line 416, `_RawPrepareStepData` is not defined as total, therefore we may set keys beyond what's defined there. In other words, this field should not be needed at all.

Reply:
when I remove that line, I get this:
```
mypy.....................................................................Failed
- hook id: mypy
- exit code: 1

tmt/trying.py:413: error: Extra key "epel" for TypedDict "_RawPrepareStepData"  [typeddict-unknown-key]
Found 1 error in 1 file (checked 68 source files)

pyright..................................................................Failed
- hook id: pyright
- exit code: 1

WARNING: there is a new pyright version available (v1.1.357 -> v1.1.365).
Please install the new version or set PYRIGHT_PYTHON_FORCE_VERSION to `latest`

/home/iquwarah/git/tmt/tmt/trying.py
  /home/iquwarah/git/tmt/tmt/trying.py:416:13 - error: Expression of type "dict[str, str]" cannot be assigned to declared type "_RawPrepareStepData"
    "epel" is an undefined item in type "_RawPrepareStepData" (reportAssignmentType)
1 error, 0 warnings, 0 informations
```

Reply:
I see. Yeah, that's unexpected and definitely misleading. I'll try to fix this typing issue, we need `_RawPrepareInstallStepData` rather than leak its fields into `_RawPrepareStepData`.

Reply:
Ok, here's my attempt: https://github.com/teemtee/tmt/pull/2986

If you rebase your branch on top of this one, you should be able to add `_RawPrepareFeatureStepData` typed dict to `prepare/feature`, and then use this new type in your code for `data`, and set `epel` key without any issue.